pkg/controllers/hpa/hpa_controller.go

```diff
@@ -140,3 +140,24 @@ func (c *HorizontalPodAutoscalerController) getTargetPlacement(objRef autoscalin
 func (c *HorizontalPodAutoscalerController) SetupWithManager(mgr controllerruntime.Manager) error {
 	return controllerruntime.NewControllerManagedBy(mgr).For(&autoscalingv1.HorizontalPodAutoscaler{}).Complete(c)
 }
+
+func (c *HorizontalPodAutoscalerController) deleteWorks(namespace, name string) error {
```

Comment:
The current method declaration, `namespace` and `name`  looks like signs for `work`, can modify the declaration of the method to:
```
deleteWorks(name string)
```

and use like:
```
deleteWorks(names.GenerateWorkName(util.HorizontalPodAutoscalerKind, name, namespace))
```